Core/PacketIO: Enabled and updated Mail [WIP]

opcod3 opened the pull request:
DO NOT MERGE THIS YET!

It was intended for 6.1.0 and I have not had the chance to test it on 6.1.2 since certain other op-codes have not been updated.

opcod3 committed:
Core/PacketIO: Enabled and updated Mail

Gigi1237 commented on src/server/game/Handlers/MailHandler.cpp:
```diff
@@ -582,8 +582,10 @@ void WorldSession::HandleGetMailList(WorldPackets::Mail::MailGetList& packet)
         // skip deleted or not delivered (deliver delay not expired) mails
-        if (m->state == MAIL_STATE_DELETED || cur_time < m->deliver_time)
+        if (m->state == MAIL_STATE_DELETED || cur_time < m->deliver_time){
+            response.TotalNumRecords--;
```
Is this a hack? It feels like one.

joschiwald replied:
this one should be better
```c++
WorldPackets::Mail::MailListResult response;
time_t curTime = time(nullptr);

for (Mail* m : player->GetMails())
{
    // skip deleted or not delivered (deliver delay not expired) mails
    if (m->state == MAIL_STATE_DELETED || curTime < m->deliver_time)
        continue;

    // max. 50 mails can be sent
    if (response.Mails.size() < 50)
        response.Mails.emplace_back(m, player);

    ++response.TotalNumRecords;
}
```

jackpoz commented on src/server/game/Handlers/MailHandler.cpp:
```diff
@@ -585,11 +584,11 @@ void WorldSession::HandleGetMailList(WorldPackets::Mail::MailGetList& packet)
         // max. 50 mails can be sent
-        if (response.Mails.size() >= 50)
-            break;
+        if (response.Mails.size() < 50)
```
are the changes in this file required by this PR or are they just refactoring/codestyle changes ?

Gigi1237 replied:
They are required changes.

Before TotalNumRecords included pending mails, this makes sure it does not. Making it include pending mails causes the client to complain about the mailbox being full.

jackpoz replied:
so that's actually a fix and should be applied to branch 335 too ?

Gigi1237 replied:
I don't know if it applies to 3.3.5. Never worked on it.

joschiwald replied:
not required on 335, in that code it was handled correctly

Carbenium commented:
Could you please remove the opcode value changes since they aren't really up to date anymore ;)

gigatotem committed:
Spells/Shaman:

Bloodlust and Heroism should also check for the Fatigued debuff from Nether Ray Hunter Pets "Netherwinds" ability.

gigatotem committed:
Spells/Mage:

Time Warp should also check for the Fatigued debuff given when a Hunter's Nether Ray pet casts Netherwinds.

MitchesD committed:
Scripts/Misc: small changes after recent commits

Naios committed:
Core/Database: Prepare statements at reconnect.
* Closes #14430

MitchesD committed:
Merge pull request #14461 from gigatotem/6.x

[6.x] Heroism/Bloodlust/Time Warp

click committed:
Core/Script: Fix non-PCH build

(cherry picked from commit 058457dda9227f50c2ea8ab66769694602aa797a)

Conflicts:
	src/server/scripts/Kalimdor/zone_bloodmyst_isle.cpp

Shauren committed:
Core/PacketIO: Updated values of all known server opcodes

Carbenium committed:
Merge pull request #14459 from Gigi1237/6.1.2_Opcodes

Core/PacketIO: Sync with WPP

opcod3 committed:
Core/PacketIO: Enabled and updated Mail
